### ISSUE TITLE
Panics while using a mounted swift

### ISSUE THREAD

**Issue by acalatrava** [rclone/rclone#5522]
#### What is the problem you are having with rclone?
While using rclone to mount a swift object storage (from OVH) i got a lot of panics:
```
2021/08/10 14:46:30 fuse: panic in handler for Setattr [ID=0x2d4de Node=0xc92 Uid=0 Gid=0 Pid=31986] size=0 mtime=2021-08-10 14:46:30.536098396 +0000 UTC handle=INVALID-0x0 lockowner: runtime error: invalid memory address or nil pointer dereference
goroutine 218215 [running]:
bazil.org/fuse/fs.(*Server).serve.func2(0x21a98b8, 0xc001465dc0, 0xc0019b0ed8, 0xc0019b0e06)
	bazil.org/fuse@v0.0.0-20200524192727-fb710f7dfd05/fs/serve.go:994 +0x19e
panic(0x1b21100, 0x2da1b00)
	runtime/panic.go:965 +0x1b9
github.com/rclone/rclone/vfs.(*File).Truncate(0xc000c97380, 0x0, 0x0, 0x0)
	github.com/rclone/rclone/vfs/file.go:765 +0x266
github.com/rclone/rclone/cmd/mount.(*File).Setattr(0xc000ccb940, 0x21b9ed0, 0xc00080e2c0, 0xc001465dc0, 0xc001440be0, 0xc00148ac01, 0x50)
	github.com/rclone/rclone/cmd/mount/file.go:57 +0x265
bazil.org/fuse/fs.(*Server).handleRequest(0xc000787340, 0x21b9ed0, 0xc00080e2c0, 0x2186f80, 0xc000ccb940, 0xc001260800, 0x21a98b8, 0xc001465dc0, 0xc0019b0ed8, 0x1, ...)
	bazil.org/fuse@v0.0.0-20200524192727-fb710f7dfd05/fs/serve.go:1082 +0x2e48
bazil.org/fuse/fs.(*Server).serve(0xc000787340, 0x21a98b8, 0xc001465dc0)
	bazil.org/fuse@v0.0.0-20200524192727-fb710f7dfd05/fs/serve.go:1015 +0x390
bazil.org/fuse/fs.(*Server).Serve.func1(0xc000787340, 0x21a98b8, 0xc001465dc0)
	bazil.org/fuse@v0.0.0-20200524192727-fb710f7dfd05/fs/serve.go:512 +0x6f
created by bazil.org/fuse/fs.(*Server).Serve
	bazil.org/fuse@v0.0.0-20200524192727-fb710f7dfd05/fs/serve.go:510 +0x354
```

#### What is your rclone version (output from `rclone version`)
```
rclone v1.56.0
- os/version: debian 8.8 (64 bit)
- os/kernel: 3.16.0-4-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.16.5
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
```
root@dev:~# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

#### Which cloud storage system are you using? (e.g. Google Drive)
OVH 

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
It's a mounted fs with this command:
```
rclone mount remote:data/ /home/test/ --vfs-cache-mode=full --allow-non-empty --allow-other --dir-cache-time 5m --no-modtime --retries 3 --vfs-cache-max-size 1G --write-back-cache --attr-timeout 5s
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
What it looks like is that `o` is `nil` here

https://github.com/rclone/rclone/blob/bb6b44d199b9341cda0610eb5dc14203b009b36e/vfs/file.go#L765

`o` can only be `nil` here if it is a file that has just been created.

Can you reproduce this easily?

Can you try this fix?

[v1.57.0-beta.5614.395928db8.fix-5522-vfs-crash](https://beta.rclone.org/branch/fix-5522-vfs-crash/v1.57.0-beta.5614.395928db8.fix-5522-vfs-crash/) on branch [fix-5522-vfs-crash](https://github.com/rclone/rclone/tree/fix-5522-vfs-crash) (uploaded in 15-30 mins)

**Comment by acalatrava** [comment]
It seems that did the trick. It's been more than 12h running with the fix and I don't see any new panics :)
